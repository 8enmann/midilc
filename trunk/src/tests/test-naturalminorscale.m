/* Test naturalminorscale.m
   Testing natural minor scale creation.  Library function 
   @author Ye
   */

main() {
  Note a;
  Number n;
  Sequence b;
  b = naturalminorscale(C);
  
  play(b);

}

naturalminorscale(root) {
	Sequence seq;
	Number n;
	seq = new_sequence();
	
	/* natural minor starts on the 6th note of the major scale in the octave below */
	root = root .+ (0 - 3);
	seq = seq + root;

	root = root .+ 2;
	seq = seq + root;
		
	root = root .+ 1;
	seq = seq + root;
	
	for(n = 1; n < 3; n=n+1)
	{
		root = root .+ 2;
		seq = seq + root;
	}
	
	root = root .+ 1;
	seq = seq + root;
	
	for(n = 1; n < 3; n=n+1)
	{
		root = root .+ 2;
		seq = seq + root;
	}

	
	return seq;
}
