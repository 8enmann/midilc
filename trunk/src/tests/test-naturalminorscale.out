0,4,57
4,4,59
8,4,60
12,4,62
16,4,64
20,4,65
24,4,67
28,4,69

